src/git.ts

```diff
@@ -51,56 +37,72 @@ async function gitRemoteURL(repoDirectory: string, remoteName: string): Promise<
     return stdout
 }
 
+interface RemoteName {
+    /**
+     * Remote name of the upstream repository,
+     * or the first remote name if no upstream is found
+     */
+    remoteName: string
+}
+
+interface Branch {
+    /**
+     * Remote branch name, or 'HEAD' if it isn't found because
+     * e.g. detached HEAD state, upstream branch points to a local branch
+     */
+    branch: string
+}
+
 /**
- * Returns the remote URL.
+ * Returns the remote name and branch
+ *
+ * @param repoDirectory the repository root directory
  */
-async function gitDefaultRemoteURL(repoDirectory: string): Promise<string> {
-    const [remote] = await gitRemoteBranch(repoDirectory)
-    if (remote !== '') {
-        return gitRemoteURL(repoDirectory, remote)
+async function gitRemoteNameAndBranch(repoDirectory: string): Promise<RemoteName & Branch> {
+    let remoteName = ''
```

Comment:
Can we also change the pattern of using empty strings to using `undefined`? I.e. making a property optional if it can be optional, otherwise just relying on control flow analysis to make sure this is assigned.
Since a branch doesn't need to have an upstream I guess it would be optional? Or should the whole object be potentially `undefined`? I can't tell if this function name should read as "get the git remote name and get the branch name" or "get the git remote name and get the **remote** branch name"

Reply:
Right now we only use the branch if the remote is found, so we should return `undefined` instead of optional properties. This would also make our type simpler, since `fileRelative` is also dependent on the remote as well. We can revisit this function if it ever has more callers. 

As for how to read the function name, I believe it's ideally the latter, but we fall back to `'HEAD'` if we can't find the remote branch name.